egs/librispeech/ASR/conformer_ctc/conformer.py

```diff
@@ -115,15 +155,28 @@ def run_encoder(
         x = self.encoder_embed(x)
         x, pos_emb = self.encoder_pos(x)
         x = x.permute(1, 0, 2)  # (B, T, F) -> (T, B, F)
-        mask = encoder_padding_mask(x.size(0), supervisions)
-        if mask is not None:
-            mask = mask.to(x.device)
-        x = self.encoder(x, pos_emb, src_key_padding_mask=mask)  # (T, B, F)
+        src_key_padding_mask = encoder_padding_mask(x.size(0), supervisions)
+        if src_key_padding_mask is not None:
+            src_key_padding_mask = src_key_padding_mask.to(x.device)
+
+        if chunk_size < 0:
+            # non-streaming decoding
+            x = self.encoder(
+                x, pos_emb, src_key_padding_mask=src_key_padding_mask
```

Comment:
This is actually incorrect. You should have used the attention mask, not the padding mask.

Reply:
i am not fully understand. which line?